app/assets/javascripts/compute_resource.js

```diff
@@ -244,27 +244,84 @@ function allocation_switcher(element, action) {
   return false;
 }
 
-function vsphereGetResourcePools(item) {
+function vsphereClusterChanged(item) {
+  var handleRequest = function(request, items) {
+      for (var i = 0; i < request.length; i++) {
+        var option = request[i].name;
+        $('<option>').text(option).val(option).appendTo(items);
+      }
+  };
+
+  var handleDatastoreRequest = function(request, items) {
+    var data = request['datastores'];
+    for (var i = 0; i < data.length; i++) {
+      var id = data[i]['datastore'].name;
+      var text = datastoreStats(data[i]['datastore']);
+      $('<option>').text(text).val(id).appendTo(items);
+    }
+  };
+
+  vsphereGetClusterItems(item, 'select[id*=resource_pool]', 'poolurl', __('Error loading resource pools: %s'), handleRequest);
+  vsphereGetClusterItems(item, 'select[id*=datastore]', 'datastoreurl', __('Error loading datastores: %s'), handleDatastoreRequest);
+  vsphereGetClusterItems(item, 'select.vmware_network', 'networkurl', __('Error loading networks: %s'), handleRequest);
+}
+
+function vsphereGetClusterItems(item, selector, datafield, errormsg, requesthandler) {
   var data = {cluster_id: $(item).val()};
-  var url = $(item).data('url');
+  var url = $(item).data(datafield);
+  var selectboxes = $(selector);
+  selectboxes.indicator_show();
   foreman.tools.showSpinner();
```

Comment:
The spinner gets hidden too early, i.e. when the first request finishes. I'd prefer removing the global spinner completely and using only the "inline" spinners.